src/utils.php

```diff
@@ -32,3 +32,15 @@ function pathToUri(string $filepath): string {
     $filepath = implode('/', array_map('urlencode', explode('/', $filepath)));
     return 'file:///' . $filepath;
 }
+
+/**
+ * Transforms URI into file path
+ * 
+ * @param string $uri
+ * @return string
+ */
+function uriToPath(string $uri)
+{
+    $filepath = urldecode(parse_url($uri)['path']);
+    return strpos($filepath, ':') === false ? $filepath : str_replace('/', '\\', $filepath);
```

Comment:
Forget what I said. the directory separator should of course be determined by the URI, not the runtime OS.